daemon-tests/tests/happy_path.rs

```diff
@@ -506,36 +508,29 @@ async fn retry_rollover_an_open_cfd() {
     )
     .await;
 
-    let taker_commit_txid_after_first_rollover = taker.latest_commit_txid();
-
-    // We ensure that the taker has a *different* commit_txid after the first rollover
-    assert_ne!(
-        taker_commit_txid_after_contract_setup,
-        taker_commit_txid_after_first_rollover,
-        "The commit_txid of the taker after the first rollover should not be the same as after contract setup"
-    );
-
-    let maker_commit_txid_after_first_rollover = maker.latest_commit_txid();
-
-    // We ensure that maker and taker have the same commit_txid after the first rollover
-    assert_eq!(
-        taker_commit_txid_after_first_rollover, maker_commit_txid_after_first_rollover,
-        "The maker and taker should have the same commit_txid after the first rollover"
-    );
+    // We simulate the taker being one rollover behind by setting the
+    // latest DLC to the one generated by contract setup
+    taker
+        .simulate_previous_rollover(
+            order_id,
+            taker_dlc_after_contract_setup,
+            taker_complete_fee_after_contract_setup,
+        )
+        .await;
 
-    // We mock the initial oracle event-id for the second rollover again
+    // We mock the initial oracle event-id for the rollover retry again
     mock_oracle_announcements(
         &mut maker,
         &mut taker,
         contract_setup_oracle_data_announcement.clone(),
     )
     .await;
```

Comment:
I would like to have a better comment here as to why this is necessary. I feel like it's just a quirk of the test, but I might be missing something! cc @da-kami

Reply:
To some extent it is a quirk, because we only have two events that we currently use in the test, i.e. we use `exmaple_0` for contract setup, then `example_1` for the first rollover, and `example_0` for the second one. I originally opted for this so I can assert that the value changes, i.e. the mocked event is correctly picked when rolling over from a previous commit. But re-using `example_0` is indeed confusing and should be changed.